### DISCUSSION TITLE
Understanding $ functions

### DISCUSSION THREAD

**Discussion by gdelmas** [facebook/lexical#2062] [Q&A] [answered]
I understand that functions starting with a `$` are to be used only in "callbacks" like updates, etc, as they need to access the active editor and state.

From an API perspective I am curious to understand the reason why [`$createOffsetView(editor: LexicalEditor,...`](https://github.com/facebook/lexical/blob/main/packages/lexical-offset/src/index.js#L467) has `editor` as an argument. Could it not read the `activeEditor` from the closure like other `$` utils do?

I am also curious to understand why other functions like `$isTextNode` start with a `$`.
```
function $isTextNode(node) {
  return node instanceof TextNode;
}
```
From the implementation it looks like this specific function could work anywhere.

It would be great if you could share the rationale for what `$` function mean.

**Comment by Proful** [comment, marked as answer]
May be you already gone through the below link
https://lexical.dev/docs/faq

**Reply by trueadm** [reply]
The $ prefix has little to do about what the function uses, and much more about where the function is useful. If you're in an update or read closure, all helper functions should be prefixed with $. 

As for why some functions take editor abs others do not? That's because they aren't in core and do not have access to getActiveEditor.

**Reply by gdelmas** [reply]
Them not being in Core explains that. Is there no possibility to expose [`getActiveEditor`](https://github.com/facebook/lexical/blob/e27b5691a7540f528efb3232f0432b65e5cae92f/packages/lexical/src/LexicalUpdates.js#L92) externally to non-core packages? From an API perspective that would align with expectations of $-functions.
